src/main/java/io/percy/selenium/Percy.java

```diff
@@ -55,27 +58,35 @@ public Percy(WebDriver driver) {
     }
 
     /**
-     * Attempts to load percy-agent.js from the resources in this Jar. The file
-     * comes from the node module @percy/agent, which is installed and packaged into
-     * this Jar as part of the Maven build.
+     * Attempts to load percy-agent.js from `http://localhost:5338/percy-agent.js`.
      *
-     * Bundling the percy-agent.js file with this library does run the minor risk of
-     * a future incompatibility between the bundled percy-agent.js in this library,
-     * and the version of @percy/agent being run by the library's client.
-     *
-     * An alternative to consider would be to try to load percy-agent.js at runtime
-     * from a running percy agent server on the standard port.
+     * This JavaScript is critical for capturing snapshots. It serializes and captures
+     * the DOM. Without it, snapshots cannot be captured.
      */
     @Nullable
     private String loadPercyAgentJs() {
+        HttpClient httpClient = HttpClientBuilder.create().build();
+
         try {
-            InputStream stream = getClass().getClassLoader().getResourceAsStream(AGENTJS_FILE);
-            byte[] agentBytes = new byte[stream.available()];
-            stream.read(agentBytes);
-            return new String(agentBytes);
-        } catch (Exception e) {
-            LOGGER.log(Level.WARNING, "Something went wrong trying to load {}. Snapshotting will not work.",
-                    AGENTJS_FILE);
+            //Creating a HttpGet object
+            HttpGet httpget = new HttpGet("http://localhost:5338/percy-agent.js");
```

Comment:
Fully acknowledge this is hard coded -- we need to make a pass through all the SDKs to enable changing the POST URL / port

Reply:
As a user, I want to load JS files from npm repository or GitHub, to skip hosting server by own.

Reply:
In order for Percy to work at all `http://localhost:5338/` has to be available (it's the server that's started when running `percy exec`). It's the same server that the DOM snapshot is `POST`'ed to for asset discovery / sending to our API. 

We do have plans to customize this URL / port, but we need to make a pass through all our SDKs to make that possible.